code/modules/client/preferences/icon_size.dm

```diff
@@ -0,0 +1,18 @@
+/datum/preference/numeric/icon_size
+	category = PREFERENCE_CATEGORY_GAME_PREFERENCES
+	savefile_key = "icon_size"
+	savefile_identifier = PREFERENCE_PLAYER
+
+	minimum = 16
+	maximum = 256 // Oh god, increase this if we start gaming with 16k monitors.
+
+	step = 32
```

Comment:
should use defines, also why? doesn't this already exist as zoom? isn't icon-size deprecated?